src/main/java/it/pn/frontend/e2e/utility/NotificationBuilder.java

```diff
@@ -0,0 +1,166 @@
+        Assert.assertNotNull("La chiamata per il preload del documento non è andata a buon fine", response);
+        log.info("PreLoad del documento effettuato con successo");
+        if (costiNotifica.equalsIgnoreCase("false")) {
+            F24Payment f24 = new F24Payment(sha256, response.get(0).getKey(), "v1", false);
+            restNotification.uploadDocument(response.get(i).getUrl(), response.get(i).getSecret(), sha256);
+            if (!payments.get(i).getPagoPa().getNoticeCode().isEmpty()) {
+                payments.get(i).setF24(f24);
+            } else {
+                NotificationPaymentItem payment = new NotificationPaymentItem(f24);
+                payments.add(payment);
+            }
+        } else {
+            F24Payment f24 = new F24Payment(sha256, response.get(0).getKey(), "v1");
+            restNotification.uploadDocument(response.get(i).getUrl(), response.get(i).getSecret(), sha256);
+            if (!payments.get(i).getPagoPa().getNoticeCode().isEmpty()) {
+                payments.get(i).setF24(f24);
+            } else {
+                NotificationPaymentItem payment = new NotificationPaymentItem(f24);
+                payments.add(payment);
+            }
+        }
+        return payments;
+    }
+
+    public ArrayList<Document> preloadDocument(int numeroDocumenti) {
+        List<PreLoadRequest> preLoadRequestList = new ArrayList<>();
+        List<String> sha256List = new ArrayList<>();
+        listForPreloadPopulation(numeroDocumenti, sha256List, preLoadRequestList);
+        List<PreLoadResponse> response = restNotification.preLoadDocument(preLoadRequestList);
+        Assert.assertNotNull("La chiamata per il preload del documento non è andata a buon fine", response);
+        log.info("PreLoad del documento effettuato con successo");
+        ArrayList<Document> documents = new ArrayList<>();
+        for (int i = 0; i < numeroDocumenti; i++) {
+            documents.add(new Document(new Digests(sha256List.get(i)), APPLICATION_PDF, new Ref(response.get(i).getKey(), "v1")));
+            restNotification.uploadDocument(response.get(i).getUrl(), response.get(i).getSecret(), sha256List.get(i));
+        }
+        return documents;
+    }
+
+    public void listForPreloadPopulation(int documentNumber, List<String> sha256List, List<PreLoadRequest> preLoadRequestList) {
+        File documentFile = new File("src/test/resources/dataPopulation/fileUpload/sample.pdf");
+        for (int i = 0; i < documentNumber; i++) {
+            String sha256 = computeSha256(documentFile.getAbsolutePath().replace("\\", "/"));
+            PreLoadRequest preLoadRequest = new PreLoadRequest()
+                    .preloadIdx("0")
+                    .sha256(sha256)
+                    .contentType(APPLICATION_PDF);
+            sha256List.add(sha256);
+            preLoadRequestList.add(preLoadRequest);
+        }
+    }
+
+    //come eccezione ho: IOException e FileNotFoundException
```

Comment:
delete this comment and use both Exception that you receive in catch

Reply:
FileNotFounException extends IOException, so i used only that